src/components/Pill/PillButton/__tests__/PillButton.spec.tsx

```diff
@@ -8,52 +10,78 @@ describe('<PillButton> component', () => {
     const onClearMock = jest.fn();
     const name = 'Pill name';
     const content = 'This pill is in use';
+    const downArrowLabel = 'down arrow';
+    const upArrowLabel = 'up arrow';
+    const clearLabel = 'clear label';
 
-    let wrapper;
+    let view: RenderResult;
+
+    const getButtonByName = (inputName) => {
+        return screen.getByRole('button', { name: `${inputName}` });
+    };
 
     afterEach(() => {
         jest.resetAllMocks();
     });
 
     describe('in inactive, collapsed state (with minimal props)', () => {
         beforeEach(() => {
-            wrapper = mount(
-                <PillButton toggleDropdown={toggleDropdownMock} onClear={onClearMock} name={name} />
+            view = render(
+                <PillButton
+                    toggleDropdown={toggleDropdownMock}
+                    onClear={onClearMock}
+                    name={name}
+                    downArrowLabel={downArrowLabel}
+                    upArrowLabel={upArrowLabel}
+                    clearLabel={clearLabel}
+                />
             );
         });
 
         it('should render correctly', () => {
-            expect(toJson(wrapper)).toMatchSnapshot();
-            expect(wrapper.find('.PillButton__pill--isActive')).toHaveLength(0);
-            expect(wrapper.find('.PillButton__pill--isOpen')).toHaveLength(0);
+            expect(view.container).toMatchSnapshot();
+            expect(getButtonByName(downArrowLabel)).not.toHaveClass(
+                'PillButton__button PillButton__button--isOpen'
+            );
+            expect(getButtonByName(downArrowLabel)).not.toHaveClass(
+                'PillButton__pill PillButton__pill--isOpen PillButton__pill--isActive'
+            );
         });
-        it('should trigger toggle state once when clicked', () => {
-            wrapper.find('.PillButton__pill').simulate('click');
+
+        it('should trigger toggle state once when clicked', async () => {
+            const user = userEvent.setup();
+            await user.click(getButtonByName(downArrowLabel));
+
             expect(toggleDropdownMock).toHaveBeenCalledTimes(1);
```

Comment:
maybe can add here a check that now the down-arrow is gone and the up-arrow is there.

Reply:
sure